pkg/didcomm/common/service/service.go

```diff
@@ -32,6 +46,33 @@ type DIDCommMsg struct {
 	OutboundDestination *Destination
 	// ToVerKeys are recipient keys
 	ToVerKeys []string
+	Common    *Common
+}
+
+// ThreadID returns msg ~thread.thid if there is no ~thread.thid returns msg @id
+// message is invalid if ~thread.thid exist and @id is absent
+func (m *DIDCommMsg) ThreadID() (string, error) {
+	if m.Common == nil {
+		if err := json.Unmarshal(m.Payload, &m.Common); err != nil {
```

Comment:
This should be done when the DIDCommMsg is created and not in the accessor.

See https://github.com/hyperledger/aries-framework-go/blob/3dfb648a8ed5745ecb57dcd946de965e8a5a0d87/pkg/framework/context/context.go#L99

Reply:
I like a constructor but not in this case :)
* too many dependencies (`Type, Payload, ToVerKeys, etc.`)
* we will execute unmarshal event it is no need

Reply:
The Unmarshal is being performed using this struct: https://github.com/hyperledger/aries-framework-go/blob/3dfb648a8ed5745ecb57dcd946de965e8a5a0d87/pkg/framework/context/context.go#L88

Couldn't we add ID and decorator.Thread to it and, hence, re-use the existing Unmarshal?

Reply:
Done!